### PR TITLE
completes Q6 get-comment-by-article-id

### PR DESCRIPTION
**Pull request by robbochobbo**
added /api/articles/:article_id/comments endpoint

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -176,6 +176,49 @@ describe('/api/articles', () => {
...
 
+describe('/api/articles/:article_id/comments', () => {
+    test('GET 200: responds with an array of comments for given article_id with correct properties', () => {
+        return request(app)
+        .get('/api/articles/3/comments')
+        .expect(200)
+        .then(({body: {comments}}) => {
+            expect(comments.length).toBe(2)
+            comments.forEach((comment) => {
+                expect(typeof comment.comment_id).toBe('number')
```
Comment (nc-saima-khanom) [inline __tests__/app.test.js:187]: Another matcher that may be useful to look into: .toMatchObject()

File: __tests__/app.test.js [hunk 2]
```diff
@@ -176,6 +176,49 @@ describe('/api/articles', () => {
...
+                expect(typeof comment.votes).toBe('number')
+                expect(typeof comment.created_at).toBe('string')
+                expect(typeof comment.author).toBe('string')
+                expect(typeof comment.body).toBe('string')
+                expect(typeof comment.article_id).toBe('number')
+            }) 
+        })
+    })
+    test('GET 200: responds with comments sorted by date descending', () => {
```
Comment (nc-saima-khanom) [inline __tests__/app.test.js:196]: we could also have a test for if the article id is valid but there are no associated comments for that article

File: endpoints.json [hunk 3]
```diff
@@ -42,5 +42,29 @@
           }
         ]
       }
+  },
+  "GET /api/articles/:article_id/comments": {
+    "description": "serves an array of objects of comments of a given article_id",
```
Comment (nc-saima-khanom) [inline endpoints.json:47]: we should also mention here that they are sorted in desc date order

File: models/models.js [hunk 4]
```diff
@@ -72,8 +68,32 @@ const fetchAllArticles = (sort_by='created_at',  order='DESC') => {
          })
 }
 
+const fetchCommentsByArticleId = (article_id) => {
+    if(isNaN(article_id)){
+        return Promise.reject({status: 400, msg:'Bad request'})
+    }
+    return db
+    .query(`SELECT 
+            comments.comment_id,
```
Comment (nc-saima-khanom) [inline models/models.js:77]: I think these are all of the columns! So we can use *

File: __tests__/app.test.js (hunk 1 above)
Comment (robbochobbo) [reply __tests__/app.test.js:187]: that sounds so much easier than what I did haha

File: models/models.js [hunk 5]
```diff
@@ -72,8 +68,32 @@ const fetchAllArticles = (sort_by='created_at',  order='DESC') => {
...
+            comments.votes,
+            comments.created_at,
+            comments.author,
+            comments.body,
+            comments.article_id
+            FROM comments
+            WHERE comments.article_id = $1
+            ORDER BY created_at DESC;`, [article_id])
+    .then((body) => {
+        if(body.rows.length === 0){
```
Comment (nc-saima-khanom) [inline models/models.js:87]: if there are no comments returned, are we sure that this means the article id doesnt exist? what about articles that havent been commented on yet?

**Pull request merged**